tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -836,6 +837,92 @@ def fully_connected(inputs,
 
 
 @add_arg_scope
+def layer_norm(inputs,
+               center=True,
+               scale=False,
+               epsilon=0.001,
+               activation_fn=None,
+               reuse=None,
+               variables_collections=None,
+               outputs_collections=None,
+               trainable=True,
+               scope=None):
+  """Adds a Layer Normalization layer from https://arxiv.org/abs/1607.06450.
+
+    "Layer Normalization"
+
+    Jimmy Lei Ba, Jamie Ryan Kiros, Geoffrey E. Hinton
+
+  Can be used as a normalizer function for conv2d and fully_connected.
+
+  Args:
+    inputs: a tensor of size `[batch_size, height, width, channels]`
+            or `[batch_size, channels]`.
+    center: If True, subtract `beta`. If False, `beta` is ignored.
+    scale: If True, multiply by `gamma`. If False, `gamma` is
+      not used. When the next layer is linear (also e.g. `nn.relu`), this can be
+      disabled since the scaling can be done by the next layer.
+    epsilon: small float added to variance to avoid dividing by zero.
```

Comment:
@mikowals That's what the tf.batch_normalization() op does (which I use here). Thinking this through a bit more:

* This can only occur if all values are identical.
* If all values are identical then the variance is zero since we first subtract the mean.
* Therefore the value of epsilon is unimportant since the end result of division will be zero.

Is there a flaw in my logic or some corner case on certain hardware I'm not considering? If not then I think the correct thing to do here is to remove the parameter and pass in a sane default to the batch_normalization op.